### PR TITLE
Auto-configure origin issuer access

### PR DESCRIPTION
**Pull request by jhiemstrawisc**
A note -- I tried but failed to figure out how to use `mapstructure` to get a clean unmarshal of the data from our config. To get this moving, I just grabbed the values manually.

Closes issue #96

### CODE REVIEW

**Comment by jhiemstrawisc** [comment]
@bbockelm, I wasn't totally confident about the scope of the assigned issue, so let me know if I'm off the mark on anything.

File: generate/param_generator.go [hunk 1]
```diff
@@ -102,14 +103,16 @@ func GenParamEnum() {
...
+			stringSliceParamMap[name] = rawName
 		case "int":
 			intParamMap[name] = rawName
 		case "bool":
 			boolParamMap[name] = rawName
 		case "duration":
 			durationParamMap[name] = rawName
 		default:
-			errMsg := "UnknownType, add a new struct and return method to the generator or add it to one of the already handles types"
+			errMsg := fmt.Sprintf("UnknownType '%s', add a new struct and return method to the generator or add it to one of the already handles types", pType)
```
Comment (turetske) [inline generate/param_generator.go:115]: Noticed a grammar error, probably mine originally.  Change `handles` to `handled`

File: generate/param_generator.go [hunk 2]
```diff
@@ -218,14 +222,16 @@ func GenParamStruct() {
...
+			goType = "[]string"
 		case "int":
 			goType = "int"
 		case "bool":
 			goType = "bool"
 		case "duration":
 			goType = "time.Duration"
 		default:
-			errMsg := "UnknownType, add a new struct and return method to the generator or add it to one of the already handles types"
+			errMsg := fmt.Sprintf("UnknownType '%s', add a new struct and return method to the generator or add it to one of the already handles types", pType)
```
Comment (turetske) [inline generate/param_generator.go:234]: Same grammar error as above

File: xrootd/authorization_test.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ func TestWriteOriginScitokensConfig(t *testing.T) {
 	err = os.WriteFile(scitokensCfg, []byte(toMergeOutput), 0640)
 	require.NoError(t, err)
 
-	err = WriteOriginScitokensConfig()
+	err = WriteOriginScitokensConfig([]string{"/foo/bar"})
```
Comment (turetske) [inline xrootd/authorization_test.go:156]: Can you add a test that checks that the ScitokensConfig file looks like you except it to with this added? Or create a test that will change what the file looks like and check that it is what is expected?

File: xrootd/xrootd_config.go [hunk 4]
```diff
@@ -263,7 +263,9 @@ to export the directory /mnt/foo to the path /bar in the data federation`)
 		return err
 	}
 
-	if err := WriteOriginScitokensConfig(); err != nil {
+	// Set up exportedPaths, which we later use to grant access to the origin's issuer.
+	exportedPaths := viper.GetStringSlice("Origin.NamespacePrefix")
```
Comment (turetske) [inline xrootd/xrootd_config.go:267]: Is this specific to origins, will a cache setup care about this?

**Review by turetske — CHANGES_REQUESTED** [review]
I have a few questions and some minor changes and tests I would like to see.

File: xrootd/xrootd_config.go (hunk 4 above)
Comment (jhiemstrawisc) [reply xrootd/xrootd_config.go:267]: I'm not sure. How does a cache handle setting up the scitokens config? Does this concept map to caches?

Comment (bbockelm) [reply xrootd/xrootd_config.go:267]: The caches will get their authorization config information from the namespace entries in the director -- it's going to look completely different from how the origins do this.

Comment (jhiemstrawisc) [reply xrootd/xrootd_config.go:267]: @turetske is there a change with this you'd like to see?

Comment (turetske) [reply xrootd/xrootd_config.go:267]: @jhiemstrawisc Nope. I was just confused.

File: xrootd/authorization_test.go [hunk 5]
```diff
@@ -153,7 +169,7 @@ func TestWriteOriginScitokensConfig(t *testing.T) {
 	err = os.WriteFile(scitokensCfg, []byte(toMergeOutput), 0640)
 	require.NoError(t, err)
 
-	err = WriteOriginScitokensConfig()
+	err = WriteOriginScitokensConfig([]string{})
```
Comment (turetske) [inline xrootd/authorization_test.go:172]: Can you add a test which does a write with a non-empty array and test that the file generated contains what is expected?

Or am I missing seeing that being tested somewhere?

Comment (jhiemstrawisc) [reply xrootd/authorization_test.go]: Okay, with latest push both the generation of the `Issuer` struct and the file writing is tested.

**Review by turetske — APPROVED** [review]
LGTM!
